src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -173,43 +178,82 @@ private Future<Void> validateLocationRestrictions(CompositePoLine poLine, List<F
    * The method checking fund location against valid locations and holding locations in POL to identify restricted locations
    * <br> if there is one, will be stored to put in error as parameter.
    * <br> otherwise, order can be opened
-   * @param poLine poLine of order that requested to be open
+   *
+   * @param poLine          poLine of order that requested to be open
    * @param restrictedFunds restricted funds of poLine
-   * @param requestContext requestContext
+   * @param requestContext  requestContext
    * @return Restricted locations
    */
   private Future<Set<String>> extractRestrictedLocationIds(CompositePoLine poLine, Set<Fund> restrictedFunds, RequestContext requestContext) {
-    Set<String> validLocationIds = poLine.getLocations().stream().map(Location::getLocationId).filter(Objects::nonNull).collect(Collectors.toSet());
-    List<String> holdingIds = poLine.getLocations().stream().map(Location::getHoldingId).filter(Objects::nonNull).toList();
-
-    return getPermanentLocationIdsFromHoldings(holdingIds, requestContext)
-      .map(permanentLocationIds -> {
-        logger.info("extractRestrictedLocations:: '{}' restrictedFund(s) is being checked against permanentLocationIds: {} and validLocations: {}",
-          restrictedFunds.size(), permanentLocationIds, validLocationIds);
-        validLocationIds.addAll(permanentLocationIds);
-        Set<String> restrictedLocationsIds = new HashSet<>();
-        for (var fund : restrictedFunds) {
-          List<String> restrictedFundLocationIds = fund.getLocationIds().stream().filter(locationId -> !validLocationIds.contains(locationId)).toList();
-          if (restrictedFundLocationIds.size() == fund.getLocationIds().size()) {
+
+    String currentTenantId = TenantTool.tenantId(requestContext.getHeaders());
+
+    var locationsForCurrentTenant = getCurrentTenantLocations(poLine, currentTenantId);
+
+    var futures = getLocationsFromHoldingsFutures(poLine, requestContext, currentTenantId);
+
+    return GenericCompositeFuture.all(futures).map(ar -> {
```

Comment:
Following lines uses features list to combine them into single result.

Reply:
I mean it will be more easy to understand to reader
getLocationsFromHoldingsFutures()
   .compose(...)
We are declaring futures variables, for me it is not clear why getLocationsFromHoldingsFutures returning furues not locations and it is not clear what types of future is

the next refactoring remove line space in line 191. 

Hopefully it will be more clean and next person easily understand code:)